### DISCUSSION TITLE
Wake on lan from any agent on the same group/network

### DISCUSSION THREAD

**Discussion by rodrigolzd** [Ylianst/MeshCentral#5547] [Ideas]
Would it be possible and if so usable, to have a wake on lan command sent to a device from any agent on the same device group network/subnet?

here is my thought, in a remote software agent only setup, I noticed one of the clients machines i need to work on is off, so to turn it on either go physically there or use wake on lan, execpt, I dont have that option on that network as the server is not on the premise but online. I cant go there as the business is closed for the day, but there is a server that is always online, tipically, windows server with active directory. 

We have the option to send a wake up call on the meshcentral interface, but that would only work on the same network (to the best of my knowledge, I could be wrong), however, if I could hit a button that would make any agent on that same device group, network or subnet send that magic packet to that target machine, I would be able to get that computer turned on.

There would be no need to have all agents send the packets, but at least one in the group. Even if I didnt look up the device mac or ip, the other agent would be able to get that info from the server, which already has all the last known IPs for that machine.

The only output needed to be recorded/displayed in mesh central would be that the packet was sent as the machine would be turned on and the machine agent would connect to the server again. 

I honestly just had this idea and have not polished it much, so I like to hear what you guys think

**Comment by si458** [comment]
meshcentral does exactually as you say 'We have the option to send a wake up call on the meshcentral interface, but that would only work on the same network'
you simple click 'actions' then select 'wake-up', and any device thats in the SAME GROUP AND ON THE SAME NETWORK/SUBNET as the device you have selected will send the 'wake up' broadcast/packets to the device to wake it up

the best way to get this to work is to test it locally first!
so use one device to try waking up another device with an application like wakemeonlan - nirsoft
once you have proved this works, then meshcentral should just work also!

the is no way for devices in the same group that ARENT on the same subnet to wake up the device you selected because its broadcast packets that get sent i believe

but i also do believe that meshcentral DOESNT record logs when you ask a device to WAKE UP or even what DEVICE was used to WAKE UP the device you selected? so you could open an issue/enhancement for us to add this in!
